spawn_worker/src/win32.rs

```diff
@@ -0,0 +1,205 @@
+            },
+            Threading::GetCurrentProcess,
+        },
+    },
+};
+
+pub enum Target {
+    Entrypoint(crate::Entrypoint),
+    ManualTrampoline(String, String),
+    Noop,
+}
+
+pub struct SpawnWorker {
+    stdin: Option<Stdio>,
+    stderr: Option<Stdio>,
+    stdout: Option<Stdio>,
+    target: Target,
+    env_clear: bool,
+}
+
+impl Default for SpawnWorker {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl SpawnWorker {
+    pub fn new() -> Self {
+        Self {
+            stdin: None,
+            stdout: None,
+            stderr: None,
+            target: Target::Noop,
+            env_clear: false,
+        }
+    }
+
+    pub fn target<T: Into<Target>>(&mut self, target: T) -> &mut Self {
+        self.target = target.into();
+        self
+    }
+
+    pub fn stdin<T: Into<Stdio>>(&mut self, stdio: T) -> &mut Self {
+        self.stdin = Some(stdio.into());
+        self
+    }
+
+    pub fn stdout<T: Into<Stdio>>(&mut self, stdio: T) -> &mut Self {
+        self.stdout = Some(stdio.into());
+        self
+    }
+
+    pub fn stderr<T: Into<Stdio>>(&mut self, stdio: T) -> &mut Self {
+        self.stderr = Some(stdio.into());
+        self
+    }
+
+    pub fn spawn(&mut self) -> anyhow::Result<Child> {
+        self.do_spawn()
+    }
+
+    fn do_spawn(&mut self) -> anyhow::Result<Child> {
+        let (f, path) = write_trampoline()?.keep()?;
+        drop(f);
+
+        let mut cmd = Command::new(path);
+
+        if let Some(stdin) = self.stdin.take() {
+            cmd.stdin(stdin);
+        }
+        if let Some(stdout) = self.stdout.take() {
+            cmd.stdout(stdout);
+        }
+
+        if let Some(stderr) = self.stderr.take() {
+            cmd.stderr(stderr);
+        }
+
+        // if self.env_clear {
+        //     cmd.env_clear();
+        // }
+
+        match &self.target {
+            Target::Entrypoint(f) => {
+                let (path, symbol_name) = get_trampoline_target_data(f.symbol_name.as_ptr() as *const u8)?;
+                cmd.args([path, symbol_name])
+            }
+            Target::ManualTrampoline(path, symbol_name) => cmd.args([path, symbol_name]),
+            Target::Noop => todo!(),
+        };
+
+        Ok(cmd.spawn()?)
+    }
+}
+
+fn get_module_file_name(h: HMODULE) -> anyhow::Result<String> {
+    let mut buf = Vec::new();
+    buf.resize(2000, 0);
+    loop {
+        let read: usize = unsafe { GetModuleFileNameW(h, &mut buf) } as usize;
+        if read == 0 {
+            return Err(unsafe { GetLastError() }
+                .ok()
+                .err()
+                .map(|e| e.into())
+                .unwrap_or_else(|| anyhow::anyhow!("unknown error getting module name")));
+        }
+
+        if read == buf.len() {
+            buf.resize(buf.len() * 2, 0)
+        } else {
+            return Ok(String::from_utf16(&buf[0..read])?);
+        }
+    }
+}
+
+fn get_sym_name(f: *const u8) -> anyhow::Result<String> {
```

Comment:
This should be removed - because currently we don't look up the trampoline symbol programatically